extensions/wikia/AttributeServiceMWApi/AttributeServiceMWApiController.class.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+class AttributeServiceMWApiController extends WikiaController {
+
+    /**
+     * Helper function for the attribute service to clear user MediaWiki.
+     * This is called when a client other than MW updates an attribute
+     * using the attribute service to clear the User cache.
+     */
+    public function purgeUserCache() {
+
+        $userId = $this->request->getVal( 'userId' );
+        $signature = $this->request->getVal( 'signature' );
+
+       if ( !$this->assertValidParameters( $userId, $signature ) ) {
+           return;
+       }
+
+        if ( !$this->assertValidSignature( $userId, $signature ) ) {
+            return;
+        }
+
+        User::clearUserCache( $userId );
```

Comment:
This method will only clear the local wiki cache.

Call `User::invalidateCache` instead as it clear shared cache from external user as well.